contracts/Example2.sol

```diff
@@ -2,10 +2,14 @@ pragma solidity ^0.8.4;
 
 contract Example2 {
     uint public counter;
- 
-    function incrementBy(uint[] memory arr) public {
-        for (uint idx = 0; idx < arr.length; idx++){
+
+    function incrementBy(uint[] calldata arr) external {
+        uint len = arr.length;
+        for (uint idx = 0; idx < len; ) {
             counter += arr[idx];
```

Comment:
Save counter in a memory variable. Here you are accessing it from storage multiple times

Reply:
Can you also take a look at the failing tests?

Reply:
Fixed it